docs/azure/partials/common/_dns-resource-group.mdx

```diff
@@ -1 +1 @@
-The `dns-azure-resource-group` is only required if using Azure as your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned.
+The `dns-azure-resource-group` flag is optional and only if using Azure are your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned.
```

Comment:
I'd rephrase to say 

"The `dns-azure-resource-group` flag is optional and only applies if Azure is your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned."

Is it required or optional if Azure is your DNS provider, if using Azure makes it no longer optional I'd say

"The `dns-azure-resource-group` flag is only required if Azure is your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned."

Reply:
Is this better?

The flag is only used if Azure is the DNS provider (ie, ignored for Cloudflare). It's an optional flag even if Azure is used. See `resource_group_name` in the [Terraform docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/dns_zone#argument-reference) as this is doing the same thing (it's actually the inspiration for the functionality)